### ISSUE TITLE
Feature Request:  Ability to specify exact number of warnings

### ISSUE THREAD

**Issue by rplankenhorn** [eslint/eslint#9738]
Version:  3.19.0

Problem:  On my current project, we use the ```max-warnings``` argument to specify the maximum warnings that are acceptable and intend to fix these later.  The problem is that occasionally, we accidentally fix a warning  and forget to update the value.  Then, someone else introduces a new warning later and the count goes up but since the threshold wasn't updated, it doesn't throw an error.

Solution:  Add an argument called ```exact-warnings``` or ```warnings```.  Eslint would throw an error if the number of warnings doesn't exactly match the specified value.  PR checks would fail based on this value and force the dev to update the number of warnings to the new value.

**Comment by platinumazure** [comment]
Hi @rplankenhorn, thanks for the issue!

I'm wondering if there might be a better way to address your use case that doesn't require a change to ESLint core/CLI, but I'm not 100% sure. I'd like to see what the rest of the team thinks.

**Comment by rplankenhorn** [comment]
@platinumazure Currently for our CI checks, I am parsing out the max-warnings using python and manually throwing an error if the value is different.  The problem with this approach is that it doesn't work for packages we pull in using lerna.  It also seems like a small change to eslint although I haven't looked at the code closely.

**Comment by not-an-aardvark** [comment]
Would it be possible to use the [`CLIEngine.executeOnFiles`](https://eslint.org/docs/developer-guide/nodejs-api#executeonfiles) API to accomplish this? You could run ESLint on your files and perform arbitrary operations on the results (e.g. matching against specific warnings) without adding the options to ESLint's CLI itself.

**Comment by rplankenhorn** [comment]
@not-an-aardvark I'll take a look!

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

That said, I think it should be possible to use ESLint's Node.js API to accomplish this without requiring changes to ESLint itself.
